src/pages/index.js

```diff
@@ -61,35 +64,83 @@ const cardTemplate = document.querySelector("#card-template");
 const cardTemplateElement = cardTemplate.content.querySelector(".card");
 const cardsList = document.querySelector(".cards__list");
 
+function handleDeleteSubmit(evt) {
+  evt.preventDefault();
+
+  const submitBtn = evt.submitter;
+  submitBtn.textContent = "Deleting...";
+
+  api
+    .removeCard(selectedCardId)
+    .then(() => {
+      selectedCard.remove();
+      closeModal(deleteModal);
+      selectedCard = null;
+      selectedCardId = null;
+    })
+    .catch(console.error)
+    .finally(() => {
+      submitBtn.textContent = "Delete";
+    });
+}
+
+function handleDeleteCard(cardElement, data) {
+  selectedCard = cardElement;
+  selectedCardId = data._id || data.id;
+  openModal(deleteModal);
+}
+
+let selectedCard;
+let selectedCardId;
+
+function handleLike(evt) {
+  const likeButton = evt.target;
+  const card = likeButton.closest(".card");
+  const cardId = card.dataset.cardId;
+  const isLiked = likeButton.classList.contains("card__like-button_liked");
+
+  api.changeLikeStatus(cardId, isLiked)
+    .then(() => {
+      likeButton.classList.toggle("card__like-button_liked");
+    })
+    .catch((err) => {
+      console.error("Error updating like status:", err);
+    });
+}
+
 function getCardElement(data) {
   const cardElement = cardTemplateElement.cloneNode(true);
   const cardNameEl = cardElement.querySelector(".card__title");
   const cardImageEl = cardElement.querySelector(".card__image");
   const likeButton = cardElement.querySelector(".card__like-button");
-  const deleteButton = cardElement.querySelector(".card__delete-button"); //delete button
+  const deleteButton = cardElement.querySelector(".card__delete-button");
 
   cardNameEl.textContent = data.name;
   cardImageEl.src = data.link;
   cardImageEl.alt = data.name;
+  cardElement.dataset.cardId = data._id || data.id;
+
```

Comment:
* When I reload the page, all **red** hearts become **transparent** again.


Please, check `isLiked` here